core/network/impl/gossiper_broadcast.cpp

```diff
@@ -62,6 +67,17 @@ namespace kagome::network {
           });
     };
 
+    for (const auto &stream : syncing_streams_) {
+      if (stream && !stream->isClosed()) {
+        msg_send_lambda(stream);
+      } else {
+        // remove this stream
+        syncing_streams_.erase(
```

Comment:
I seems it's bad idea to modify vector during iterate by elements of them.
I offer to separate it to two stages: writing to stream and wiping of closed.

Reply:
will remove as here https://www.techiedelight.com/remove-elements-vector-inside-loop-cpp/

Reply:
https://en.cppreference.com/w/cpp/container/vector/erase
> Invalidates iterators and references at or after the point of the erase, including the end() iterator.
And `range-for` cycle use `end()` internally, in this case it would be invalidated.